### PR TITLE
Include unsubscribe link in failure/recovery email

### PR DESCRIPTION
**Pull request by tejohnso**
The email templates are now unique per addressee and display id, so we can't send
multiple recipients the same email.

### CODE REVIEW

File: test/unit/notifier.js [hunk 1]
```diff
@@ -91,7 +91,7 @@ describe("Notifier - Unit", () => {
 
       assert.equal(parameters.from, "support@risevision.com");
       assert.equal(parameters.fromName, "Rise Vision Support");
-      assert.deepEqual(parameters.recipients, ['a@example.com', 'b@example.com']);
+      assert.deepEqual(parameters.recipients, 'b@example.com');
```
Comment (santiagonoguez) [inline test/unit/notifier.js]: equal instead of deepEqual

File: src/notifier.js [hunk 2]
```diff
@@ -64,30 +64,34 @@ function sendRecoveryEmail(displayId, addresses) {
 function prepareAndSendEmail(template, displayId, recipients) {
   const subject = template.subject.replace('DISPLAYID', displayId);
   const text = template.body.replace(/DISPLAYID/g, displayId);
+  const promises = [];
+
+  recipients.forEach(recipient=>{
```
Comment (andrecardoso) [inline src/notifier.js]: A `map` would fit here, but I'm OK with this approach.

File: src/templates/failure.txt [hunk 3]
```diff
@@ -1 +1,3 @@
 The display with display id 'DISPLAYID' has gone offline.
+
+If you no longer wish to receive these notices, you can <a href="http://rvaserver2/monitoring/unsubscribe?email=EMAIL&id=DISPLAYID>unsubscribe</a>.
```
Comment (alexey-rise) [inline src/templates/failure.txt]: @tejohnso is `rvaserver2` a placeholder?

File: src/notifier.js [hunk 4]
```diff
@@ -64,30 +64,34 @@ function sendRecoveryEmail(displayId, addresses) {
...
-  const data = {
-    from: SENDER_ADDRESS,
-    fromName: SENDER_NAME,
-    recipients,
-    subject,
-    text
-  };
-
-  return send(data);
+  return Promise.all(promises);
```
Comment (andrecardoso) [inline src/notifier.js]: Is it OK to fail fast here? Some recipients might not receive the notification if there's a problem in one of them

File: src/notifier.js [hunk 5]
```diff
@@ -64,30 +64,34 @@ function sendRecoveryEmail(displayId, addresses) {
...
 }
 
 function send(data) {
   const parameterString = querystring.stringify(data);
   const url = `${EMAIL_API_URL}?${parameterString}`;
 
   return got.post(url)
-  .then(response =>
-  {
+  .then(response => {
```
Comment (andrecardoso) [inline src/notifier.js]: 👏

File: src/notifier.js (hunk 4 above)
Comment (santiagonoguez) [reply src/notifier.js]: Interesting point, as any rejection will automatically cause return a failed promise. I wonder if non-rejected promises are guaranteed to finish even in that case, or they are not completed.

File: src/templates/failure.txt (hunk 3 above)
Comment (tejohnso) [reply src/templates/failure.txt]: I meant to revisit the url after doing some testing, but then I noticed the PR on core hasn't been merged. Can you merge that?  Then I'll do some testing to confirm it works and I'll update this url.

Comment (alexey-rise) [reply src/templates/failure.txt]: @tejohnso merged.

File: src/notifier.js (hunk 4 above)
Comment (tejohnso) [reply src/notifier.js]: All of the promises are scheduled when the array is constructed.
Promise.all will fail fast on the first rejected promise, but I don't think it is going to cancel execution of all of the remaining unresolved promises.

Comment (tejohnso) [reply src/notifier.js]: Confirmed.

Comment (andrecardoso) [reply src/notifier.js]: Thanks. 👍

**Pull request merged**